### PR TITLE
Use the Request in Forwarder directly

### PR DESCRIPTION
**Pull request by mgencur**
This is a follow-up to https://github.com/knative-extensions/reconciler-test/pull/572 . This PR doesn't choose specific headers but forwards everything by re-using the original request and modifying it.

# Changes

-
-
-

/kind <kind>

Fixes #

**Release Note**

```release-note

```

**Docs**

```docs

```

### CODE REVIEW

**Review by ReToCode — APPROVED** [review]
That looks even better, thanks @mgencur 

/lgtm

File: pkg/eventshub/forwarder/forwarder.go [hunk 1]
```diff
@@ -165,17 +161,15 @@ func (o *Forwarder) ServeHTTP(writer http.ResponseWriter, request *http.Request)
 		logging.FromContext(o.ctx).Fatalw("Error while venting the received event", zap.Error(err))
 	}
 
-	req, err := http.NewRequestWithContext(requestCtx, http.MethodPost, o.Sink, nil)
-	if err != nil {
-		logging.FromContext(o.ctx).Error("Cannot create the request: ", err)
-	}
-
-	req.Header = headersToBeSent
+	req := request.Clone(requestCtx)
```
Comment (matzew) [inline pkg/eventshub/forwarder/forwarder.go:164]: I like this clone approach

**Review by matzew — APPROVED** [review]
/lgtm
/approve

**Comment by ReToCode** [comment]
@mgencur cherry-pick to 1.10 and 1.11 or go with what we have there?

**Comment by mgencur** [comment]
I would not bother but we can cherry-pick. I'm fine with both.

**Pull request merged**